src/lib.rs

```diff
@@ -693,3 +904,35 @@ fn black_box(x: Errno) -> Errno {
     core::sync::atomic::compiler_fence(core::sync::atomic::Ordering::SeqCst);
     x
 }
+
+#[repr(C)]
+pub enum Descriptor {
+    Closed,
+    File(File),
+    Log,
+}
+
+#[repr(C)]
+pub struct File {
+    fd: wasi_filesystem::Descriptor,
+    position: u64,
+}
+
+impl Descriptor {
+    fn get(fd: Fd) -> &'static mut Descriptor {
+        unsafe {
+            let mut fds = replace_fds(null_mut());
+            if fds.is_null() {
+                fds = core::arch::wasm32::memory_grow(0, 1) as *mut u8;
```

Comment:
I think that you'll want to check `-1` for the return value and additionally multiply the return value by the page size to get the final pointer